backend/mlarchive/archive/views.py

```diff
@@ -824,4 +824,54 @@ def get_context_data(self, **kwargs):
             date = date.replace(day=1)
         kwargs['subscribers'] = Subscriber.objects.filter(date=date)
         kwargs['date'] = date
-        return kwargs
\ No newline at end of file
+        return kwargs
+
+
+class ReportsMessagesView(View):
+    """Message Counts Report"""
+    
+    def get_message_stats(self, sdate, edate):
+        """Returns a tuple ( total messages, message counts as
+        list of tuples (listname, count))"""
+        messages = Message.objects.filter(
+            date__gte=sdate,
+            date__lt=edate,
+            email_list__private=False)
+        counter = Counter(messages.values_list('email_list__name', flat=True))
+        return (messages.count(), sorted(counter.items()))
+
+    def get(self, request, *args, **kwargs):
+        form = DateForm(request.GET)
+        
+        # if no date submitted default to last month
+        if not request.GET:
+            today = datetime.date.today()
+            edate = today.replace(day=1)
+            sdate = edate - relativedelta(months=1)
+            total, message_counts = self.get_message_stats(sdate, edate)
+            form = DateForm(initial={
+                'start_date': sdate.strftime('%Y-%m-%d'),
+                'end_date': edate.strftime('%Y-%m-%d'),
+            })
+        elif form.is_valid():
+            sdate = form.cleaned_data['start_date']
+            edate = form.cleaned_data['end_date']
+            total, message_counts = self.get_message_stats(sdate, edate)
+            # always pass back an unbound form to avoid annoying is-valid styling
```

Comment:
This looks like an opportunity to change how you're holding it?

Reply:
Not sure I understand this comment

Reply:
That you have to wipe the form to avoid the view behavior (styling) with the form is a sign that your form and view are not built to work well together, or that the styling needs adjusting.

Reply:
This stems from a number of circumstances. First, the page flow, which doesn't redirect on successful form submission but instead returns the same page with new date parameters, and potentially a bound form. Django-bootstrap, using server side validation, adds class "is-valid" to fields with no errors, which bootstrap CSS styles with green border and checkbox. It would be best not to have this styling unless it is to distinguish between other fields that do have errors. I considered other solutions: disable server side validation (but this is site wide), don't use {% bootstrap_field field %} tags, but that would require custom styling to mimic the bootstrap styling of other forms. Passing an unbound form seems like a reasonable workaround for this situation.

Reply:
Well, that's exactly it - you're fighting assumptions from the framework about how the framework will be used, but I'll relent. We can talk about general patterns for this kind of thing in person. (Passing an unbound form may be the right thing to do for other reasons than just styling too).